src/org/loklak/objects/Timeline2.java

```diff
@@ -0,0 +1,457 @@
+/**
+ *  Timeline2
+ *  Copyright 22.02.2015 by Michael Peter Christen, @0rb1t3r
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *  
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *  
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program in the file lgpl21.txt
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.loklak.objects;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.loklak.data.DAO;
+import org.loklak.data.DAO.IndexName;
+import org.loklak.harvester.Post;
+import org.loklak.susi.SusiThought;
+
+/**
+ * A timeline2 is a structure which holds tweet for the purpose of presentation
+ * There is no tweet retrieval method here, just an iterator which returns the tweets in reverse appearing order
+ */
+//TODO: remove MessageEntry
+public class Timeline2 implements Iterable<Post> {
+
+    public static enum Order {
+        CREATED_AT("date"),
+        TIMESTAMP("long"),
```

Comment:
timestamp is no. of sec from 1 jan 1970, used for creating PostId, one can use it to create date from it.